### ISSUE TITLE
App Fatal Error

### ISSUE THREAD

**Issue by BannorPhil** [EDDiscovery/EDDiscovery#2136]
Had updated all star systems, closed the app then ran the new installer. Installed successfully, but get this when I run it:

An unhandled fatal exception has occurred.

==== BEGIN ====
System.Data.SQLite.SQLiteException (0x80004005): SQL logic error
no such table: main.EdsmSystems
   at System.Data.SQLite.SQLite3.Prepare(SQLiteConnection cnn, String strSql, SQLiteStatement previous, UInt32 timeoutMS, String& strRemain)
   at System.Data.SQLite.SQLiteCommand.BuildNextCommand()
   at System.Data.SQLite.SQLiteDataReader.NextResult()
   at System.Data.SQLite.SQLiteDataReader..ctor(SQLiteCommand cmd, CommandBehavior behave)
   at System.Data.SQLite.SQLiteCommand.ExecuteReader(CommandBehavior behavior)
   at System.Data.SQLite.SQLiteCommand.ExecuteNonQuery(CommandBehavior behavior)
   at EliteDangerousCore.DB.SQLiteCommandED`1.ExecuteNonQuery() in C:\Code\EDDiscovery\EliteDangerous\DB\SQLiteCommandED.cs:line 559
   at EliteDangerousCore.DB.SQLiteConnectionSystem.CreateSystemsTableIndexesNoLock() in C:\Code\EDDiscovery\EliteDangerous\DB\SQLiteConnectionSystem.cs:line 341
   at EliteDangerousCore.DB.SQLiteConnectionSystem.CreateSystemsTableIndexes() in C:\Code\EDDiscovery\EliteDangerous\DB\SQLiteConnectionSystem.cs:line 318
   at EDDiscovery.EDDiscoveryController.BackgroundInit() in C:\Code\EDDiscovery\EDDiscovery\EDDiscoveryController.cs:line 552
   at EDDiscovery.EDDiscoveryController.BackgroundWorkerThread() in C:\Code\EDDiscovery\EDDiscovery\EDDiscoveryController.cs:line 495
   at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state, Boolean preserveSyncCtx)
   at System.Threading.ExecutionContext.Run(ExecutionContext executionContext, ContextCallback callback, Object state, Boolean preserveSyncCtx)
   at System.Threading.ExecutionContext.Run(ExecutionContext executionContext, ContextCallback callback, Object state)
   at System.Threading.ThreadHelper.ThreadStart()
===== END =====

Philip Bannor

**Comment by klightspeed** [comment]
If you open `EDDSystems.sqlite` with something like SQLiteBrowser, do the `EdsmSystems_temp` and/or `SystemNames_temp` tables exist?

It sounds like the application was forcefully terminated between dropping `EdsmSystems` and renaming `EdsmSystems_temp` to `EdsmSystems`.

It looks like SQLite transactions don't work for drop and rename statements.

**Comment by robbyxp1** [comment]
To fix your issue, delete in the appfolder eddsystems.sqlite (either manually or via safe mode - hold down shift when you run EDD) and it will rebuild it.

**Comment by AndySpawn** [comment]
I'm in the same boat here.

An unhandled fatal exception has occurred.

==== BEGIN ====
System.Data.SQLite.SQLiteException (0x80004005): SQL logic error
no such table: main.EdsmSystems
   at System.Data.SQLite.SQLite3.Prepare(SQLiteConnection cnn, String strSql, SQLiteStatement previous, UInt32 timeoutMS, String& strRemain)
   at System.Data.SQLite.SQLiteCommand.BuildNextCommand()
   at System.Data.SQLite.SQLiteDataReader.NextResult()
   at System.Data.SQLite.SQLiteDataReader..ctor(SQLiteCommand cmd, CommandBehavior behave)
   at System.Data.SQLite.SQLiteCommand.ExecuteReader(CommandBehavior behavior)
   at System.Data.SQLite.SQLiteCommand.ExecuteNonQuery(CommandBehavior behavior)
   at EliteDangerousCore.DB.SQLiteCommandED`1.ExecuteNonQuery() in C:\Code\EDDiscovery\EliteDangerous\DB\SQLiteCommandED.cs:line 559
   at EliteDangerousCore.DB.SQLiteConnectionSystem.CreateSystemsTableIndexesNoLock() in C:\Code\EDDiscovery\EliteDangerous\DB\SQLiteConnectionSystem.cs:line 341
   at EliteDangerousCore.DB.SQLiteConnectionSystem.CreateSystemsTableIndexes() in C:\Code\EDDiscovery\EliteDangerous\DB\SQLiteConnectionSystem.cs:line 318
   at EDDiscovery.EDDiscoveryController.BackgroundInit() in C:\Code\EDDiscovery\EDDiscovery\EDDiscoveryController.cs:line 552
   at EDDiscovery.EDDiscoveryController.BackgroundWorkerThread() in C:\Code\EDDiscovery\EDDiscovery\EDDiscoveryController.cs:line 495
   at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state, Boolean preserveSyncCtx)
   at System.Threading.ExecutionContext.Run(ExecutionContext executionContext, ContextCallback callback, Object state, Boolean preserveSyncCtx)
   at System.Threading.ExecutionContext.Run(ExecutionContext executionContext, ContextCallback callback, Object state)
   at System.Threading.ThreadHelper.ThreadStart()
===== END =====

I've tried your fix, robbyxp1, but no luck.  I've had to go back to v10.1.6

It may be nothing, but my Database file is called EDDSystem.sqlite not EDDSystems.sqlite

Also, I don't have either EdsmSystems_temp or SystemNames_temp tables
![screenshot 2018-08-29 22 08 39](https://user-images.githubusercontent.com/23283728/44815835-8526d380-abd8-11e8-8a60-81a0ba1f4d3e.png)

**Comment by klightspeed** [comment]
If you are using the options.txt file to redirect the data directory, 10.2 has a bug that ignores that file.

**Comment by AndySpawn** [comment]
I wasn't, no.

I also tried rebuilding the database in 10.1.6 and then installing 10.2.0, but this didn't change anything.

Also, If I launch 10.2.0 in Safe Mode and try to move the Database, or Delete/Rebuild Database, it prompts me with the wrong location every time.  It keeps saying that the database is located at F:/Program when it's actually at F:\Program Files (x86)\EDDiscovery\EDDiscovery\Database because the Fatal Error occurs every time I Run EDDiscovery.

**Comment by klightspeed** [comment]
This is a related bug in 10.2 - paths with spaces from `options.txt` or `dboptions.txt` were being split at the first space.  #2139 should fix this.

**Comment by AndySpawn** [comment]
No worries.  Just trying to give as much info as I can.

**Comment by AndySpawn** [comment]
Ok, now 10.1.6 has stopped working too.

An unhandled fatal exception has occurred.

==== BEGIN ====
System.Data.SQLite.SQLiteException (0x80004005): SQL logic error
no such table: main.EdsmSystems
   at System.Data.SQLite.SQLite3.Prepare(SQLiteConnection cnn, String strSql, SQLiteStatement previous, UInt32 timeoutMS, String& strRemain)
   at System.Data.SQLite.SQLiteCommand.BuildNextCommand()
   at System.Data.SQLite.SQLiteDataReader.NextResult()
   at System.Data.SQLite.SQLiteDataReader..ctor(SQLiteCommand cmd, CommandBehavior behave)
   at System.Data.SQLite.SQLiteCommand.ExecuteReader(CommandBehavior behavior)
   at System.Data.SQLite.SQLiteCommand.ExecuteNonQuery(CommandBehavior behavior)
   at EliteDangerousCore.DB.SQLiteCommandED`1.ExecuteNonQuery() in C:\Code\EDDiscovery\EliteDangerous\DB\SQLiteCommandED.cs:line 548
   at EliteDangerousCore.DB.SQLiteConnectionSystem.CreateSystemsTableIndexesNoLock() in C:\Code\EDDiscovery\EliteDangerous\DB\SQLiteConnectionSystem.cs:line 341
   at EliteDangerousCore.DB.SQLiteConnectionSystem.CreateSystemsTableIndexes() in C:\Code\EDDiscovery\EliteDangerous\DB\SQLiteConnectionSystem.cs:line 318
   at EDDiscovery.EDDiscoveryController.BackgroundInit() in C:\Code\EDDiscovery\EDDiscovery\EDDiscoveryController.cs:line 552
   at EDDiscovery.EDDiscoveryController.BackgroundWorkerThread() in C:\Code\EDDiscovery\EDDiscovery\EDDiscoveryController.cs:line 495
   at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state, Boolean preserveSyncCtx)
   at System.Threading.ExecutionContext.Run(ExecutionContext executionContext, ContextCallback callback, Object state, Boolean preserveSyncCtx)
   at System.Threading.ExecutionContext.Run(ExecutionContext executionContext, ContextCallback callback, Object state)
   at System.Threading.ThreadHelper.ThreadStart()
===== END =====

**Comment by AndySpawn** [comment]
Ok, 10.1.6 is back after editing the dboptions.txt file

**Comment by BannorPhil** [comment]
I red EDD in safe mode, deleted the DB. App still didn't run.
I uninstalled and manually deleted all data. Reinstalled 10.1.6.
When I ran it, it gave me:

A fatal exception was encountered while initializing EDDiscovery.

==== BEGIN ====
System.Data.SQLite.SQLiteException (0x80004005): unable to open database file
   at System.Data.SQLite.SQLite3.Open(String strFilename, String vfsName, SQLiteConnectionFlags connectionFlags, SQLiteOpenFlagsEnum openFlags, Int32 maxPoolSize, Boolean usePool)
   at System.Data.SQLite.SQLiteConnection.Open()
   at EliteDangerousCore.DB.SQLiteConnectionED`1..ctor(Nullable`1 maindb, Boolean utctimeindicator, Boolean initializing, Boolean shortlived) in C:\Code\EDDiscovery\EliteDangerous\DB\SQLiteConnectionED.cs:line 147
   at EliteDangerousCore.DB.SQLiteConnectionUser.<>c.<Initialize>b__4_0() in C:\Code\EDDiscovery\EliteDangerous\DB\SQLiteConnectionUser.cs:line 45
   at EliteDangerousCore.DB.SQLiteConnectionED`1.InitializeIfNeeded(Action initializer) in C:\Code\EDDiscovery\EliteDangerous\DB\SQLiteConnectionED.cs:line 87
   at EDDiscovery.EDDiscoveryController.InitializeDatabases() in C:\Code\EDDiscovery\EDDiscovery\EDDiscoveryController.cs:line 383
   at EDDiscovery.EDDiscoveryController.Initialize(Action`1 msg) in C:\Code\EDDiscovery\EDDiscovery\EDDiscoveryController.cs:line 145
   at EDDiscovery.EDDApplicationContext.MainEngineStart(Object sender, EventArgs e) in C:\Code\EDDiscovery\EDDiscovery\EDDApplicationContext.cs:line 137
===== END =====

Very impressed with the various responses! :)
What else should I try?

Philip Bannor
